# Calling Times API

We will extract the weekly best seller's list for the last 10 years from the Times API.

In [1]:
import numpy as np
import pandas as pd
import requests
import json
import datetime
import time
import re

from TimesExtractor import TimesExtractor

I use the manually defined class `TimesExtractor`, which will take `start_date`, `end_date`, and `frequencies` as an argument and will make calls to the Times Best Seller Books API based on the number of weeks within those dates range.

In [2]:
year = TimesExtractor(start_date='2011-05-01', end_date='2020-05-01', frq='W')
dic = year.extractor()

  0%|                                                                                          | 0/470 [00:00<?, ?it/s]

Due to API's limitation there is a lapse of 3 seconds between calls
2011-05-01


  0%|▏                                                                                 | 1/470 [00:03<26:45,  3.42s/it]

2011-05-08


  0%|▎                                                                                 | 2/470 [00:06<26:38,  3.42s/it]

2011-05-15


  1%|▌                                                                                 | 3/470 [00:10<26:38,  3.42s/it]

2011-05-22


  1%|▋                                                                                 | 4/470 [00:13<26:21,  3.39s/it]

2011-05-29


  1%|▊                                                                                 | 5/470 [00:16<26:14,  3.39s/it]

2011-06-05


  1%|█                                                                                 | 6/470 [00:20<26:20,  3.41s/it]

2011-06-12


  1%|█▏                                                                                | 7/470 [00:23<26:28,  3.43s/it]

2011-06-19


  2%|█▍                                                                                | 8/470 [00:27<26:21,  3.42s/it]

2011-06-26


  2%|█▌                                                                                | 9/470 [00:30<26:17,  3.42s/it]

2011-07-03


  2%|█▋                                                                               | 10/470 [00:34<26:20,  3.44s/it]

2011-07-10


  2%|█▉                                                                               | 11/470 [00:37<26:12,  3.43s/it]

2011-07-17


  3%|██                                                                               | 12/470 [00:40<26:04,  3.42s/it]

2011-07-24


  3%|██▏                                                                              | 13/470 [00:44<25:56,  3.41s/it]

2011-07-31


  3%|██▍                                                                              | 14/470 [00:47<25:48,  3.40s/it]

2011-08-07


  3%|██▌                                                                              | 15/470 [00:51<25:54,  3.42s/it]

2011-08-14


  3%|██▊                                                                              | 16/470 [00:54<25:43,  3.40s/it]

2011-08-21


  4%|██▉                                                                              | 17/470 [00:57<25:44,  3.41s/it]

2011-08-28


  4%|███                                                                              | 18/470 [01:01<25:53,  3.44s/it]

2011-09-04


  4%|███▎                                                                             | 19/470 [01:05<26:03,  3.47s/it]

2011-09-11


  4%|███▍                                                                             | 20/470 [01:08<25:50,  3.45s/it]

2011-09-18


  4%|███▌                                                                             | 21/470 [01:11<25:49,  3.45s/it]

2011-09-25


  5%|███▊                                                                             | 22/470 [01:15<25:41,  3.44s/it]

2011-10-02


  5%|███▉                                                                             | 23/470 [01:18<25:27,  3.42s/it]

2011-10-09


  5%|████▏                                                                            | 24/470 [01:22<25:24,  3.42s/it]

2011-10-16


  5%|████▎                                                                            | 25/470 [01:25<25:34,  3.45s/it]

2011-10-23


  6%|████▍                                                                            | 26/470 [01:29<25:24,  3.43s/it]

2011-10-30


  6%|████▋                                                                            | 27/470 [01:32<25:21,  3.43s/it]

2011-11-06


  6%|████▊                                                                            | 28/470 [01:35<25:07,  3.41s/it]

2011-11-13


  6%|████▉                                                                            | 29/470 [01:39<25:07,  3.42s/it]

2011-11-20


  6%|█████▏                                                                           | 30/470 [01:42<25:03,  3.42s/it]

2011-11-27


  7%|█████▎                                                                           | 31/470 [01:46<25:02,  3.42s/it]

2011-12-04


  7%|█████▌                                                                           | 32/470 [01:49<25:02,  3.43s/it]

2011-12-11


  7%|█████▋                                                                           | 33/470 [01:53<25:05,  3.45s/it]

2011-12-18


  7%|█████▊                                                                           | 34/470 [01:56<25:08,  3.46s/it]

2011-12-25


  7%|██████                                                                           | 35/470 [02:00<26:34,  3.67s/it]

2012-01-01


  8%|██████▏                                                                          | 36/470 [02:04<25:53,  3.58s/it]

2012-01-08


  8%|██████▍                                                                          | 37/470 [02:07<25:45,  3.57s/it]

2012-01-15


  8%|██████▌                                                                          | 38/470 [02:10<25:19,  3.52s/it]

2012-01-22


  8%|██████▋                                                                          | 39/470 [02:14<25:06,  3.49s/it]

2012-01-29


  9%|██████▉                                                                          | 40/470 [02:17<24:51,  3.47s/it]

2012-02-05


  9%|███████                                                                          | 41/470 [02:21<24:43,  3.46s/it]

2012-02-12


  9%|███████▏                                                                         | 42/470 [02:24<24:41,  3.46s/it]

2012-02-19


  9%|███████▍                                                                         | 43/470 [02:28<24:29,  3.44s/it]

2012-02-26


  9%|███████▌                                                                         | 44/470 [02:31<24:21,  3.43s/it]

2012-03-04


 10%|███████▊                                                                         | 45/470 [02:34<24:13,  3.42s/it]

2012-03-11


 10%|███████▉                                                                         | 46/470 [02:38<24:17,  3.44s/it]

2012-03-18


 10%|████████                                                                         | 47/470 [02:41<24:05,  3.42s/it]

2012-03-25


 10%|████████▎                                                                        | 48/470 [02:45<23:53,  3.40s/it]

2012-04-01


 10%|████████▍                                                                        | 49/470 [02:48<24:01,  3.42s/it]

2012-04-08


 11%|████████▌                                                                        | 50/470 [02:52<23:54,  3.42s/it]

2012-04-15


 11%|████████▊                                                                        | 51/470 [02:55<23:39,  3.39s/it]

2012-04-22


 11%|████████▉                                                                        | 52/470 [02:58<23:28,  3.37s/it]

2012-04-29


 11%|█████████▏                                                                       | 53/470 [03:01<23:19,  3.36s/it]

2012-05-06


 11%|█████████▎                                                                       | 54/470 [03:05<23:25,  3.38s/it]

2012-05-13


 12%|█████████▍                                                                       | 55/470 [03:08<23:28,  3.39s/it]

2012-05-20


 12%|█████████▋                                                                       | 56/470 [03:12<23:25,  3.39s/it]

2012-05-27


 12%|█████████▊                                                                       | 57/470 [03:15<23:23,  3.40s/it]

2012-06-03


 12%|█████████▉                                                                       | 58/470 [03:18<23:09,  3.37s/it]

2012-06-10


 13%|██████████▏                                                                      | 59/470 [03:22<23:13,  3.39s/it]

2012-06-17


 13%|██████████▎                                                                      | 60/470 [03:25<23:13,  3.40s/it]

2012-06-24


 13%|██████████▌                                                                      | 61/470 [03:29<23:06,  3.39s/it]

2012-07-01


 13%|██████████▋                                                                      | 62/470 [03:32<23:07,  3.40s/it]

2012-07-08


 13%|██████████▊                                                                      | 63/470 [03:36<23:08,  3.41s/it]

2012-07-15


 14%|███████████                                                                      | 64/470 [03:39<23:08,  3.42s/it]

2012-07-22


 14%|███████████▏                                                                     | 65/470 [03:43<23:43,  3.52s/it]

2012-07-29


 14%|███████████▎                                                                     | 66/470 [03:46<23:25,  3.48s/it]

2012-08-05


 14%|███████████▌                                                                     | 67/470 [03:50<23:22,  3.48s/it]

2012-08-12


 14%|███████████▋                                                                     | 68/470 [03:53<23:18,  3.48s/it]

2012-08-19


 15%|███████████▉                                                                     | 69/470 [03:56<22:59,  3.44s/it]

2012-08-26


 15%|████████████                                                                     | 70/470 [04:00<22:46,  3.42s/it]

2012-09-02


 15%|████████████▏                                                                    | 71/470 [04:03<22:47,  3.43s/it]

2012-09-09


 15%|████████████▍                                                                    | 72/470 [04:07<22:54,  3.45s/it]

2012-09-16


 16%|████████████▌                                                                    | 73/470 [04:10<23:09,  3.50s/it]

2012-09-23


 16%|████████████▊                                                                    | 74/470 [04:14<22:57,  3.48s/it]

2012-09-30


 16%|████████████▉                                                                    | 75/470 [04:17<22:52,  3.47s/it]

2012-10-07


 16%|█████████████                                                                    | 76/470 [04:21<22:47,  3.47s/it]

2012-10-14


 16%|█████████████▎                                                                   | 77/470 [04:24<22:45,  3.47s/it]

2012-10-21


 17%|█████████████▍                                                                   | 78/470 [04:28<22:44,  3.48s/it]

2012-10-28


 17%|█████████████▌                                                                   | 79/470 [04:31<22:47,  3.50s/it]

2012-11-04


 17%|█████████████▊                                                                   | 80/470 [04:35<22:29,  3.46s/it]

2012-11-11


 17%|█████████████▉                                                                   | 81/470 [04:38<22:30,  3.47s/it]

2012-11-18


 17%|██████████████▏                                                                  | 82/470 [04:42<22:19,  3.45s/it]

2012-11-25


 18%|██████████████▎                                                                  | 83/470 [04:45<22:16,  3.45s/it]

2012-12-02


 18%|██████████████▍                                                                  | 84/470 [04:48<22:17,  3.47s/it]

2012-12-09


 18%|██████████████▋                                                                  | 85/470 [04:52<22:05,  3.44s/it]

2012-12-16


 18%|██████████████▊                                                                  | 86/470 [04:55<22:03,  3.45s/it]

2012-12-23


 19%|██████████████▉                                                                  | 87/470 [04:59<22:14,  3.49s/it]

2012-12-30


 19%|███████████████▏                                                                 | 88/470 [05:02<22:04,  3.47s/it]

2013-01-06


 19%|███████████████▎                                                                 | 89/470 [05:06<21:53,  3.45s/it]

2013-01-13


 19%|███████████████▌                                                                 | 90/470 [05:09<21:47,  3.44s/it]

2013-01-20


 19%|███████████████▋                                                                 | 91/470 [05:13<21:42,  3.44s/it]

2013-01-27


 20%|███████████████▊                                                                 | 92/470 [05:16<21:33,  3.42s/it]

2013-02-03


 20%|████████████████                                                                 | 93/470 [05:19<21:30,  3.42s/it]

2013-02-10


 20%|████████████████▏                                                                | 94/470 [05:23<21:24,  3.42s/it]

2013-02-17


 20%|████████████████▎                                                                | 95/470 [05:26<21:21,  3.42s/it]

2013-02-24


 20%|████████████████▌                                                                | 96/470 [05:30<21:15,  3.41s/it]

2013-03-03


 21%|████████████████▋                                                                | 97/470 [05:33<21:13,  3.41s/it]

2013-03-10


 21%|████████████████▉                                                                | 98/470 [05:36<21:14,  3.43s/it]

2013-03-17


 21%|█████████████████                                                                | 99/470 [05:40<21:11,  3.43s/it]

2013-03-24


 21%|█████████████████                                                               | 100/470 [05:43<21:07,  3.43s/it]

2013-03-31


 21%|█████████████████▏                                                              | 101/470 [05:47<21:02,  3.42s/it]

2013-04-07


 22%|█████████████████▎                                                              | 102/470 [05:50<21:06,  3.44s/it]

2013-04-14


 22%|█████████████████▌                                                              | 103/470 [05:54<21:08,  3.46s/it]

2013-04-21


 22%|█████████████████▋                                                              | 104/470 [05:57<21:18,  3.49s/it]

2013-04-28


 22%|█████████████████▊                                                              | 105/470 [06:01<21:11,  3.48s/it]

2013-05-05


 23%|██████████████████                                                              | 106/470 [06:04<20:55,  3.45s/it]

2013-05-12


 23%|██████████████████▏                                                             | 107/470 [06:08<20:55,  3.46s/it]

2013-05-19


 23%|██████████████████▍                                                             | 108/470 [06:11<20:45,  3.44s/it]

2013-05-26


 23%|██████████████████▌                                                             | 109/470 [06:14<20:41,  3.44s/it]

2013-06-02


 23%|██████████████████▋                                                             | 110/470 [06:18<20:39,  3.44s/it]

2013-06-09


 24%|██████████████████▉                                                             | 111/470 [06:21<20:43,  3.46s/it]

2013-06-16


 24%|███████████████████                                                             | 112/470 [06:25<20:38,  3.46s/it]

2013-06-23


 24%|███████████████████▏                                                            | 113/470 [06:28<20:35,  3.46s/it]

2013-06-30


 24%|███████████████████▍                                                            | 114/470 [06:32<20:27,  3.45s/it]

2013-07-07


 24%|███████████████████▌                                                            | 115/470 [06:35<20:24,  3.45s/it]

2013-07-14


 25%|███████████████████▋                                                            | 116/470 [06:39<20:30,  3.48s/it]

2013-07-21


 25%|███████████████████▉                                                            | 117/470 [06:42<20:20,  3.46s/it]

2013-07-28


 25%|████████████████████                                                            | 118/470 [06:46<20:20,  3.47s/it]

2013-08-04


 25%|████████████████████▎                                                           | 119/470 [06:49<20:11,  3.45s/it]

2013-08-11


 26%|████████████████████▍                                                           | 120/470 [06:52<20:01,  3.43s/it]

2013-08-18


 26%|████████████████████▌                                                           | 121/470 [06:56<20:01,  3.44s/it]

2013-08-25


 26%|████████████████████▊                                                           | 122/470 [06:59<19:51,  3.42s/it]

2013-09-01


 26%|████████████████████▉                                                           | 123/470 [07:03<19:45,  3.42s/it]

2013-09-08


 26%|█████████████████████                                                           | 124/470 [07:06<19:42,  3.42s/it]

2013-09-15


 27%|█████████████████████▎                                                          | 125/470 [07:10<20:09,  3.50s/it]

2013-09-22


 27%|█████████████████████▍                                                          | 126/470 [07:13<19:51,  3.46s/it]

2013-09-29


 27%|█████████████████████▌                                                          | 127/470 [07:17<19:43,  3.45s/it]

2013-10-06


 27%|█████████████████████▊                                                          | 128/470 [07:20<19:30,  3.42s/it]

2013-10-13


 27%|█████████████████████▉                                                          | 129/470 [07:23<19:25,  3.42s/it]

2013-10-20


 28%|██████████████████████▏                                                         | 130/470 [07:27<19:18,  3.41s/it]

2013-10-27


 28%|██████████████████████▎                                                         | 131/470 [07:30<19:35,  3.47s/it]

2013-11-03


 28%|██████████████████████▍                                                         | 132/470 [07:34<19:35,  3.48s/it]

2013-11-10


 28%|██████████████████████▋                                                         | 133/470 [07:37<19:21,  3.45s/it]

2013-11-17


 29%|██████████████████████▊                                                         | 134/470 [07:41<19:09,  3.42s/it]

2013-11-24


 29%|██████████████████████▉                                                         | 135/470 [07:44<19:05,  3.42s/it]

2013-12-01


 29%|███████████████████████▏                                                        | 136/470 [07:47<19:06,  3.43s/it]

2013-12-08


 29%|███████████████████████▎                                                        | 137/470 [07:51<19:02,  3.43s/it]

2013-12-15


 29%|███████████████████████▍                                                        | 138/470 [07:54<18:59,  3.43s/it]

2013-12-22


 30%|███████████████████████▋                                                        | 139/470 [07:58<19:16,  3.49s/it]

2013-12-29


 30%|███████████████████████▊                                                        | 140/470 [08:01<18:58,  3.45s/it]

2014-01-05


 30%|████████████████████████                                                        | 141/470 [08:05<18:46,  3.42s/it]

2014-01-12


 30%|████████████████████████▏                                                       | 142/470 [08:08<18:54,  3.46s/it]

2014-01-19


 30%|████████████████████████▎                                                       | 143/470 [08:12<18:53,  3.47s/it]

2014-01-26


 31%|████████████████████████▌                                                       | 144/470 [08:15<18:41,  3.44s/it]

2014-02-02


 31%|████████████████████████▋                                                       | 145/470 [08:19<18:38,  3.44s/it]

2014-02-09


 31%|████████████████████████▊                                                       | 146/470 [08:22<18:37,  3.45s/it]

2014-02-16


 31%|█████████████████████████                                                       | 147/470 [08:25<18:28,  3.43s/it]

2014-02-23


 31%|█████████████████████████▏                                                      | 148/470 [08:29<18:23,  3.43s/it]

2014-03-02


 32%|█████████████████████████▎                                                      | 149/470 [08:32<18:15,  3.41s/it]

2014-03-09


 32%|█████████████████████████▌                                                      | 150/470 [08:36<18:15,  3.42s/it]

2014-03-16


 32%|█████████████████████████▋                                                      | 151/470 [08:39<18:14,  3.43s/it]

2014-03-23


 32%|█████████████████████████▊                                                      | 152/470 [08:43<18:13,  3.44s/it]

2014-03-30


 33%|██████████████████████████                                                      | 153/470 [08:46<18:03,  3.42s/it]

2014-04-06


 33%|██████████████████████████▏                                                     | 154/470 [08:49<17:59,  3.42s/it]

2014-04-13


 33%|██████████████████████████▍                                                     | 155/470 [08:53<17:52,  3.40s/it]

2014-04-20


 33%|██████████████████████████▌                                                     | 156/470 [08:56<17:48,  3.40s/it]

2014-04-27


 33%|██████████████████████████▋                                                     | 157/470 [08:59<17:44,  3.40s/it]

2014-05-04


 34%|██████████████████████████▉                                                     | 158/470 [09:03<18:31,  3.56s/it]

2014-05-11


 34%|███████████████████████████                                                     | 159/470 [09:08<19:35,  3.78s/it]

2014-05-18


 34%|███████████████████████████▏                                                    | 160/470 [09:11<19:04,  3.69s/it]

2014-05-25


 34%|███████████████████████████▍                                                    | 161/470 [09:15<18:52,  3.66s/it]

2014-06-01


 34%|███████████████████████████▌                                                    | 162/470 [09:18<18:24,  3.59s/it]

2014-06-08


 35%|███████████████████████████▋                                                    | 163/470 [09:22<18:37,  3.64s/it]

2014-06-15


 35%|███████████████████████████▉                                                    | 164/470 [09:25<18:15,  3.58s/it]

2014-06-22


 35%|████████████████████████████                                                    | 165/470 [09:29<17:53,  3.52s/it]

2014-06-29


 35%|████████████████████████████▎                                                   | 166/470 [09:32<17:39,  3.49s/it]

2014-07-06


 36%|████████████████████████████▍                                                   | 167/470 [09:36<17:37,  3.49s/it]

2014-07-13


 36%|████████████████████████████▌                                                   | 168/470 [09:39<17:51,  3.55s/it]

2014-07-20


 36%|████████████████████████████▊                                                   | 169/470 [09:43<17:34,  3.50s/it]

2014-07-27


 36%|████████████████████████████▉                                                   | 170/470 [09:46<17:22,  3.48s/it]

2014-08-03


 36%|█████████████████████████████                                                   | 171/470 [09:50<17:10,  3.45s/it]

2014-08-10


 37%|█████████████████████████████▎                                                  | 172/470 [09:54<18:14,  3.67s/it]

2014-08-17


 37%|█████████████████████████████▍                                                  | 173/470 [09:57<17:46,  3.59s/it]

2014-08-24


 37%|█████████████████████████████▌                                                  | 174/470 [10:01<17:23,  3.52s/it]

2014-08-31


 37%|█████████████████████████████▊                                                  | 175/470 [10:04<17:14,  3.51s/it]

2014-09-07


 37%|█████████████████████████████▉                                                  | 176/470 [10:09<19:50,  4.05s/it]

2014-09-14


 38%|██████████████████████████████▏                                                 | 177/470 [10:13<18:51,  3.86s/it]

2014-09-21


 38%|██████████████████████████████▎                                                 | 178/470 [10:16<18:12,  3.74s/it]

2014-09-28


 38%|██████████████████████████████▍                                                 | 179/470 [10:20<18:43,  3.86s/it]

2014-10-05


 38%|██████████████████████████████▋                                                 | 180/470 [10:24<18:01,  3.73s/it]

2014-10-12


 39%|██████████████████████████████▊                                                 | 181/470 [10:27<17:35,  3.65s/it]

2014-10-19


 39%|██████████████████████████████▉                                                 | 182/470 [10:31<17:16,  3.60s/it]

2014-10-26


 39%|███████████████████████████████▏                                                | 183/470 [10:34<17:08,  3.58s/it]

2014-11-02


 39%|███████████████████████████████▎                                                | 184/470 [10:38<16:58,  3.56s/it]

2014-11-09


 39%|███████████████████████████████▍                                                | 185/470 [10:41<16:48,  3.54s/it]

2014-11-16


 40%|███████████████████████████████▋                                                | 186/470 [10:45<16:37,  3.51s/it]

2014-11-23


 40%|███████████████████████████████▊                                                | 187/470 [10:48<16:26,  3.48s/it]

2014-11-30


 40%|████████████████████████████████                                                | 188/470 [10:52<16:17,  3.47s/it]

2014-12-07


 40%|████████████████████████████████▏                                               | 189/470 [10:55<16:13,  3.46s/it]

2014-12-14


 40%|████████████████████████████████▎                                               | 190/470 [10:59<16:17,  3.49s/it]

2014-12-21


 41%|████████████████████████████████▌                                               | 191/470 [11:02<16:05,  3.46s/it]

2014-12-28


 41%|████████████████████████████████▋                                               | 192/470 [11:05<16:02,  3.46s/it]

2015-01-04


 41%|████████████████████████████████▊                                               | 193/470 [11:09<15:50,  3.43s/it]

2015-01-11


 41%|█████████████████████████████████                                               | 194/470 [11:12<15:43,  3.42s/it]

2015-01-18


 41%|█████████████████████████████████▏                                              | 195/470 [11:16<15:35,  3.40s/it]

2015-01-25


 42%|█████████████████████████████████▎                                              | 196/470 [11:19<15:33,  3.41s/it]

2015-02-01


 42%|█████████████████████████████████▌                                              | 197/470 [11:22<15:32,  3.42s/it]

2015-02-08


 42%|█████████████████████████████████▋                                              | 198/470 [11:26<15:27,  3.41s/it]

2015-02-15


 42%|█████████████████████████████████▊                                              | 199/470 [11:29<15:31,  3.44s/it]

2015-02-22


 43%|██████████████████████████████████                                              | 200/470 [11:33<15:24,  3.42s/it]

2015-03-01


 43%|██████████████████████████████████▏                                             | 201/470 [11:36<15:19,  3.42s/it]

2015-03-08


 43%|██████████████████████████████████▍                                             | 202/470 [11:39<15:16,  3.42s/it]

2015-03-15


 43%|██████████████████████████████████▌                                             | 203/470 [11:43<15:21,  3.45s/it]

2015-03-22


 43%|██████████████████████████████████▋                                             | 204/470 [11:46<15:20,  3.46s/it]

2015-03-29


 44%|██████████████████████████████████▉                                             | 205/470 [11:50<15:11,  3.44s/it]

2015-04-05


 44%|███████████████████████████████████                                             | 206/470 [11:53<15:05,  3.43s/it]

2015-04-12


 44%|███████████████████████████████████▏                                            | 207/470 [11:57<15:01,  3.43s/it]

2015-04-19


 44%|███████████████████████████████████▍                                            | 208/470 [12:00<15:01,  3.44s/it]

2015-04-26


 44%|███████████████████████████████████▌                                            | 209/470 [12:04<14:58,  3.44s/it]

2015-05-03


 45%|███████████████████████████████████▋                                            | 210/470 [12:07<14:54,  3.44s/it]

2015-05-10


 45%|███████████████████████████████████▉                                            | 211/470 [12:10<14:49,  3.43s/it]

2015-05-17


 45%|████████████████████████████████████                                            | 212/470 [12:14<14:45,  3.43s/it]

2015-05-24


 45%|████████████████████████████████████▎                                           | 213/470 [12:17<14:39,  3.42s/it]

2015-05-31


 46%|████████████████████████████████████▍                                           | 214/470 [12:21<14:35,  3.42s/it]

2015-06-07


 46%|████████████████████████████████████▌                                           | 215/470 [12:24<14:29,  3.41s/it]

2015-06-14


 46%|████████████████████████████████████▊                                           | 216/470 [12:28<14:29,  3.42s/it]

2015-06-21


 46%|████████████████████████████████████▉                                           | 217/470 [12:31<14:23,  3.41s/it]

2015-06-28


 46%|█████████████████████████████████████                                           | 218/470 [12:34<14:21,  3.42s/it]

2015-07-05


 47%|█████████████████████████████████████▎                                          | 219/470 [12:38<14:15,  3.41s/it]

2015-07-12


 47%|█████████████████████████████████████▍                                          | 220/470 [12:41<14:11,  3.41s/it]

2015-07-19


 47%|█████████████████████████████████████▌                                          | 221/470 [12:45<14:15,  3.43s/it]

2015-07-26


 47%|█████████████████████████████████████▊                                          | 222/470 [12:48<14:12,  3.44s/it]

2015-08-02


 47%|█████████████████████████████████████▉                                          | 223/470 [12:52<14:13,  3.46s/it]

2015-08-09


 48%|██████████████████████████████████████▏                                         | 224/470 [12:55<14:13,  3.47s/it]

2015-08-16


 48%|██████████████████████████████████████▎                                         | 225/470 [12:58<14:01,  3.43s/it]

2015-08-23


 48%|██████████████████████████████████████▍                                         | 226/470 [13:02<13:54,  3.42s/it]

2015-08-30


 48%|██████████████████████████████████████▋                                         | 227/470 [13:05<13:51,  3.42s/it]

2015-09-06


 49%|██████████████████████████████████████▊                                         | 228/470 [13:09<13:46,  3.42s/it]

2015-09-13


 49%|██████████████████████████████████████▉                                         | 229/470 [13:12<13:41,  3.41s/it]

2015-09-20


 49%|███████████████████████████████████████▏                                        | 230/470 [13:16<13:47,  3.45s/it]

2015-09-27


 49%|███████████████████████████████████████▎                                        | 231/470 [13:19<13:40,  3.43s/it]

2015-10-04


 49%|███████████████████████████████████████▍                                        | 232/470 [13:22<13:38,  3.44s/it]

2015-10-11


 50%|███████████████████████████████████████▋                                        | 233/470 [13:26<13:33,  3.43s/it]

2015-10-18


 50%|███████████████████████████████████████▊                                        | 234/470 [13:29<13:27,  3.42s/it]

2015-10-25


 50%|████████████████████████████████████████                                        | 235/470 [13:33<13:31,  3.46s/it]

2015-11-01


 50%|████████████████████████████████████████▏                                       | 236/470 [13:36<13:24,  3.44s/it]

2015-11-08


 50%|████████████████████████████████████████▎                                       | 237/470 [13:40<13:15,  3.42s/it]

2015-11-15


 51%|████████████████████████████████████████▌                                       | 238/470 [13:43<13:14,  3.42s/it]

2015-11-22


 51%|████████████████████████████████████████▋                                       | 239/470 [13:46<13:09,  3.42s/it]

2015-11-29


 51%|████████████████████████████████████████▊                                       | 240/470 [13:50<13:10,  3.44s/it]

2015-12-06


 51%|█████████████████████████████████████████                                       | 241/470 [13:53<12:58,  3.40s/it]

2015-12-13


 51%|█████████████████████████████████████████▏                                      | 242/470 [13:57<12:50,  3.38s/it]

2015-12-20


 52%|█████████████████████████████████████████▎                                      | 243/470 [14:00<13:00,  3.44s/it]

2015-12-27


 52%|█████████████████████████████████████████▌                                      | 244/470 [14:04<12:58,  3.45s/it]

2016-01-03


 52%|█████████████████████████████████████████▋                                      | 245/470 [14:07<12:53,  3.44s/it]

2016-01-10


 52%|█████████████████████████████████████████▊                                      | 246/470 [14:10<12:48,  3.43s/it]

2016-01-17


 53%|██████████████████████████████████████████                                      | 247/470 [14:14<12:46,  3.44s/it]

2016-01-24


 53%|██████████████████████████████████████████▏                                     | 248/470 [14:17<12:49,  3.46s/it]

2016-01-31


 53%|██████████████████████████████████████████▍                                     | 249/470 [14:21<12:43,  3.45s/it]

2016-02-07


 53%|██████████████████████████████████████████▌                                     | 250/470 [14:24<12:39,  3.45s/it]

2016-02-14


 53%|██████████████████████████████████████████▋                                     | 251/470 [14:28<12:36,  3.46s/it]

2016-02-21


 54%|██████████████████████████████████████████▉                                     | 252/470 [14:31<12:33,  3.46s/it]

2016-02-28


 54%|███████████████████████████████████████████                                     | 253/470 [14:35<12:21,  3.42s/it]

2016-03-06


 54%|███████████████████████████████████████████▏                                    | 254/470 [14:38<12:18,  3.42s/it]

2016-03-13


 54%|███████████████████████████████████████████▍                                    | 255/470 [14:41<12:14,  3.42s/it]

2016-03-20


 54%|███████████████████████████████████████████▌                                    | 256/470 [14:45<12:15,  3.44s/it]

2016-03-27


 55%|███████████████████████████████████████████▋                                    | 257/470 [14:48<12:10,  3.43s/it]

2016-04-03


 55%|███████████████████████████████████████████▉                                    | 258/470 [14:52<11:58,  3.39s/it]

2016-04-10


 55%|████████████████████████████████████████████                                    | 259/470 [14:55<11:56,  3.40s/it]

2016-04-17


 55%|████████████████████████████████████████████▎                                   | 260/470 [14:59<12:03,  3.44s/it]

2016-04-24


 56%|████████████████████████████████████████████▍                                   | 261/470 [15:02<11:57,  3.43s/it]

2016-05-01


 56%|████████████████████████████████████████████▌                                   | 262/470 [15:05<11:49,  3.41s/it]

2016-05-08


 56%|████████████████████████████████████████████▊                                   | 263/470 [15:09<11:45,  3.41s/it]

2016-05-15


 56%|████████████████████████████████████████████▉                                   | 264/470 [15:12<11:43,  3.41s/it]

2016-05-22


 56%|█████████████████████████████████████████████                                   | 265/470 [15:15<11:36,  3.40s/it]

2016-05-29


 57%|█████████████████████████████████████████████▎                                  | 266/470 [15:19<11:32,  3.40s/it]

2016-06-05


 57%|█████████████████████████████████████████████▍                                  | 267/470 [15:22<11:30,  3.40s/it]

2016-06-12


 57%|█████████████████████████████████████████████▌                                  | 268/470 [15:26<11:29,  3.41s/it]

2016-06-19


 57%|█████████████████████████████████████████████▊                                  | 269/470 [15:29<11:29,  3.43s/it]

2016-06-26


 57%|█████████████████████████████████████████████▉                                  | 270/470 [15:33<11:23,  3.42s/it]

2016-07-03


 58%|██████████████████████████████████████████████▏                                 | 271/470 [15:36<11:21,  3.43s/it]

2016-07-10


 58%|██████████████████████████████████████████████▎                                 | 272/470 [15:39<11:16,  3.42s/it]

2016-07-17


 58%|██████████████████████████████████████████████▍                                 | 273/470 [15:43<11:15,  3.43s/it]

2016-07-24


 58%|██████████████████████████████████████████████▋                                 | 274/470 [15:46<11:18,  3.46s/it]

2016-07-31


 59%|██████████████████████████████████████████████▊                                 | 275/470 [15:50<11:11,  3.44s/it]

2016-08-07


 59%|██████████████████████████████████████████████▉                                 | 276/470 [15:53<11:06,  3.44s/it]

2016-08-14


 59%|███████████████████████████████████████████████▏                                | 277/470 [15:57<11:07,  3.46s/it]

2016-08-21


 59%|███████████████████████████████████████████████▎                                | 278/470 [16:00<10:56,  3.42s/it]

2016-08-28


 59%|███████████████████████████████████████████████▍                                | 279/470 [16:03<10:53,  3.42s/it]

2016-09-04


 60%|███████████████████████████████████████████████▋                                | 280/470 [16:07<10:50,  3.42s/it]

2016-09-11


 60%|███████████████████████████████████████████████▊                                | 281/470 [16:10<10:50,  3.44s/it]

2016-09-18


 60%|████████████████████████████████████████████████                                | 282/470 [16:14<10:48,  3.45s/it]

2016-09-25


 60%|████████████████████████████████████████████████▏                               | 283/470 [16:17<10:44,  3.45s/it]

2016-10-02


 60%|████████████████████████████████████████████████▎                               | 284/470 [16:21<10:42,  3.45s/it]

2016-10-09


 61%|████████████████████████████████████████████████▌                               | 285/470 [16:24<10:42,  3.47s/it]

2016-10-16


 61%|████████████████████████████████████████████████▋                               | 286/470 [16:28<10:32,  3.44s/it]

2016-10-23


 61%|████████████████████████████████████████████████▊                               | 287/470 [16:31<10:27,  3.43s/it]

2016-10-30


 61%|█████████████████████████████████████████████████                               | 288/470 [16:34<10:22,  3.42s/it]

2016-11-06


 61%|█████████████████████████████████████████████████▏                              | 289/470 [16:38<10:16,  3.41s/it]

2016-11-13


 62%|█████████████████████████████████████████████████▎                              | 290/470 [16:41<10:15,  3.42s/it]

2016-11-20


 62%|█████████████████████████████████████████████████▌                              | 291/470 [16:45<10:10,  3.41s/it]

2016-11-27


 62%|█████████████████████████████████████████████████▋                              | 292/470 [16:48<10:07,  3.41s/it]

2016-12-04


 62%|█████████████████████████████████████████████████▊                              | 293/470 [16:52<10:05,  3.42s/it]

2016-12-11


 63%|██████████████████████████████████████████████████                              | 294/470 [16:55<09:59,  3.41s/it]

2016-12-18


 63%|██████████████████████████████████████████████████▏                             | 295/470 [16:58<09:54,  3.40s/it]

2016-12-25


 63%|██████████████████████████████████████████████████▍                             | 296/470 [17:02<09:49,  3.39s/it]

2017-01-01


 63%|██████████████████████████████████████████████████▌                             | 297/470 [17:05<09:47,  3.40s/it]

2017-01-08


 63%|██████████████████████████████████████████████████▋                             | 298/470 [17:08<09:44,  3.40s/it]

2017-01-15


 64%|██████████████████████████████████████████████████▉                             | 299/470 [17:12<09:42,  3.41s/it]

2017-01-22


 64%|███████████████████████████████████████████████████                             | 300/470 [17:15<09:38,  3.41s/it]

2017-01-29


 64%|███████████████████████████████████████████████████▏                            | 301/470 [17:19<09:39,  3.43s/it]

2017-02-05


 64%|███████████████████████████████████████████████████▍                            | 302/470 [17:22<09:38,  3.44s/it]

2017-02-12


 64%|███████████████████████████████████████████████████▌                            | 303/470 [17:26<09:36,  3.45s/it]

2017-02-19


 65%|███████████████████████████████████████████████████▋                            | 304/470 [17:29<09:28,  3.43s/it]

2017-02-26


 65%|███████████████████████████████████████████████████▉                            | 305/470 [17:33<09:25,  3.43s/it]

2017-03-05


 65%|████████████████████████████████████████████████████                            | 306/470 [17:36<09:34,  3.50s/it]

2017-03-12


 65%|████████████████████████████████████████████████████▎                           | 307/470 [17:40<09:28,  3.49s/it]

2017-03-19


 66%|████████████████████████████████████████████████████▍                           | 308/470 [17:43<09:28,  3.51s/it]

2017-03-26


 66%|████████████████████████████████████████████████████▌                           | 309/470 [17:47<09:18,  3.47s/it]

2017-04-02


 66%|████████████████████████████████████████████████████▊                           | 310/470 [17:50<09:15,  3.47s/it]

2017-04-09


 66%|████████████████████████████████████████████████████▉                           | 311/470 [17:54<09:16,  3.50s/it]

2017-04-16


 66%|█████████████████████████████████████████████████████                           | 312/470 [17:57<09:17,  3.53s/it]

2017-04-23


 67%|█████████████████████████████████████████████████████▎                          | 313/470 [18:01<09:06,  3.48s/it]

2017-04-30


 67%|█████████████████████████████████████████████████████▍                          | 314/470 [18:04<09:01,  3.47s/it]

2017-05-07


 67%|█████████████████████████████████████████████████████▌                          | 315/470 [18:08<08:57,  3.47s/it]

2017-05-14


 67%|█████████████████████████████████████████████████████▊                          | 316/470 [18:11<08:49,  3.44s/it]

2017-05-21


 67%|█████████████████████████████████████████████████████▉                          | 317/470 [18:14<08:43,  3.42s/it]

2017-05-28


 68%|██████████████████████████████████████████████████████▏                         | 318/470 [18:18<08:37,  3.41s/it]

2017-06-04


 68%|██████████████████████████████████████████████████████▎                         | 319/470 [18:21<08:30,  3.38s/it]

2017-06-11


 68%|██████████████████████████████████████████████████████▍                         | 320/470 [18:24<08:26,  3.38s/it]

2017-06-18


 68%|██████████████████████████████████████████████████████▋                         | 321/470 [18:28<08:23,  3.38s/it]

2017-06-25


 69%|██████████████████████████████████████████████████████▊                         | 322/470 [18:31<08:17,  3.36s/it]

2017-07-02


 69%|██████████████████████████████████████████████████████▉                         | 323/470 [18:34<08:16,  3.38s/it]

2017-07-09


 69%|███████████████████████████████████████████████████████▏                        | 324/470 [18:38<08:12,  3.38s/it]

2017-07-16


 69%|███████████████████████████████████████████████████████▎                        | 325/470 [18:41<08:15,  3.42s/it]

2017-07-23


 69%|███████████████████████████████████████████████████████▍                        | 326/470 [18:48<10:25,  4.34s/it]

2017-07-30


 70%|███████████████████████████████████████████████████████▋                        | 327/470 [18:51<09:42,  4.07s/it]

2017-08-06


 70%|███████████████████████████████████████████████████████▊                        | 328/470 [18:55<09:10,  3.88s/it]

2017-08-13


 70%|████████████████████████████████████████████████████████                        | 329/470 [18:58<08:46,  3.73s/it]

2017-08-20


 70%|████████████████████████████████████████████████████████▏                       | 330/470 [19:01<08:25,  3.61s/it]

2017-08-27


 70%|████████████████████████████████████████████████████████▎                       | 331/470 [19:05<08:17,  3.58s/it]

2017-09-03


 71%|████████████████████████████████████████████████████████▌                       | 332/470 [19:08<08:08,  3.54s/it]

2017-09-10


 71%|████████████████████████████████████████████████████████▋                       | 333/470 [19:12<07:59,  3.50s/it]

2017-09-17


 71%|████████████████████████████████████████████████████████▊                       | 334/470 [19:15<07:51,  3.46s/it]

2017-09-24


 71%|█████████████████████████████████████████████████████████                       | 335/470 [19:19<07:44,  3.44s/it]

2017-10-01


 71%|█████████████████████████████████████████████████████████▏                      | 336/470 [19:22<07:36,  3.40s/it]

2017-10-08


 72%|█████████████████████████████████████████████████████████▎                      | 337/470 [19:25<07:33,  3.41s/it]

2017-10-15


 72%|█████████████████████████████████████████████████████████▌                      | 338/470 [19:29<07:32,  3.43s/it]

2017-10-22


 72%|█████████████████████████████████████████████████████████▋                      | 339/470 [19:32<07:29,  3.43s/it]

2017-10-29


 72%|█████████████████████████████████████████████████████████▊                      | 340/470 [19:36<07:24,  3.42s/it]

2017-11-05


 73%|██████████████████████████████████████████████████████████                      | 341/470 [19:39<07:20,  3.41s/it]

2017-11-12


 73%|██████████████████████████████████████████████████████████▏                     | 342/470 [19:42<07:16,  3.41s/it]

2017-11-19


 73%|██████████████████████████████████████████████████████████▍                     | 343/470 [19:46<07:13,  3.41s/it]

2017-11-26


 73%|██████████████████████████████████████████████████████████▌                     | 344/470 [19:49<07:09,  3.41s/it]

2017-12-03


 73%|██████████████████████████████████████████████████████████▋                     | 345/470 [19:53<07:04,  3.40s/it]

2017-12-10


 74%|██████████████████████████████████████████████████████████▉                     | 346/470 [19:56<07:00,  3.39s/it]

2017-12-17


 74%|███████████████████████████████████████████████████████████                     | 347/470 [19:59<06:56,  3.39s/it]

2017-12-24


 74%|███████████████████████████████████████████████████████████▏                    | 348/470 [20:03<06:52,  3.38s/it]

2017-12-31


 74%|███████████████████████████████████████████████████████████▍                    | 349/470 [20:06<06:45,  3.35s/it]

2018-01-07


 74%|███████████████████████████████████████████████████████████▌                    | 350/470 [20:09<06:41,  3.35s/it]

2018-01-14


 75%|███████████████████████████████████████████████████████████▋                    | 351/470 [20:13<06:39,  3.36s/it]

2018-01-21


 75%|███████████████████████████████████████████████████████████▉                    | 352/470 [20:16<06:35,  3.35s/it]

2018-01-28


 75%|████████████████████████████████████████████████████████████                    | 353/470 [20:19<06:33,  3.36s/it]

2018-02-04


 75%|████████████████████████████████████████████████████████████▎                   | 354/470 [20:23<06:30,  3.37s/it]

2018-02-11


 76%|████████████████████████████████████████████████████████████▍                   | 355/470 [20:26<06:29,  3.39s/it]

2018-02-18


 76%|████████████████████████████████████████████████████████████▌                   | 356/470 [20:30<06:28,  3.41s/it]

2018-02-25


 76%|████████████████████████████████████████████████████████████▊                   | 357/470 [20:33<06:24,  3.40s/it]

2018-03-04


 76%|████████████████████████████████████████████████████████████▉                   | 358/470 [20:36<06:19,  3.39s/it]

2018-03-11


 76%|█████████████████████████████████████████████████████████████                   | 359/470 [20:40<06:16,  3.39s/it]

2018-03-18


 77%|█████████████████████████████████████████████████████████████▎                  | 360/470 [20:43<06:18,  3.44s/it]

2018-03-25


 77%|█████████████████████████████████████████████████████████████▍                  | 361/470 [20:47<06:10,  3.40s/it]

2018-04-01


 77%|█████████████████████████████████████████████████████████████▌                  | 362/470 [20:50<06:10,  3.43s/it]

2018-04-08


 77%|█████████████████████████████████████████████████████████████▊                  | 363/470 [20:54<06:13,  3.49s/it]

2018-04-15


 77%|█████████████████████████████████████████████████████████████▉                  | 364/470 [20:57<06:07,  3.46s/it]

2018-04-22


 78%|██████████████████████████████████████████████████████████████▏                 | 365/470 [21:01<06:02,  3.45s/it]

2018-04-29


 78%|██████████████████████████████████████████████████████████████▎                 | 366/470 [21:04<05:57,  3.44s/it]

2018-05-06


 78%|██████████████████████████████████████████████████████████████▍                 | 367/470 [21:07<05:52,  3.43s/it]

2018-05-13


 78%|██████████████████████████████████████████████████████████████▋                 | 368/470 [21:11<05:48,  3.42s/it]

2018-05-20


 79%|██████████████████████████████████████████████████████████████▊                 | 369/470 [21:15<05:57,  3.54s/it]

2018-05-27


 79%|██████████████████████████████████████████████████████████████▉                 | 370/470 [21:18<05:47,  3.48s/it]

2018-06-03


 79%|███████████████████████████████████████████████████████████████▏                | 371/470 [21:22<05:48,  3.52s/it]

2018-06-10


 79%|███████████████████████████████████████████████████████████████▎                | 372/470 [21:25<05:39,  3.47s/it]

2018-06-17


 79%|███████████████████████████████████████████████████████████████▍                | 373/470 [21:28<05:32,  3.43s/it]

2018-06-24


 80%|███████████████████████████████████████████████████████████████▋                | 374/470 [21:32<05:27,  3.41s/it]

2018-07-01


 80%|███████████████████████████████████████████████████████████████▊                | 375/470 [21:35<05:21,  3.39s/it]

2018-07-08


 80%|████████████████████████████████████████████████████████████████                | 376/470 [21:38<05:18,  3.39s/it]

2018-07-15


 80%|████████████████████████████████████████████████████████████████▏               | 377/470 [21:42<05:14,  3.38s/it]

2018-07-22


 80%|████████████████████████████████████████████████████████████████▎               | 378/470 [21:45<05:12,  3.39s/it]

2018-07-29


 81%|████████████████████████████████████████████████████████████████▌               | 379/470 [21:49<05:09,  3.40s/it]

2018-08-05


 81%|████████████████████████████████████████████████████████████████▋               | 380/470 [21:52<05:05,  3.40s/it]

2018-08-12


 81%|████████████████████████████████████████████████████████████████▊               | 381/470 [21:55<05:02,  3.40s/it]

2018-08-19


 81%|█████████████████████████████████████████████████████████████████               | 382/470 [21:59<04:59,  3.40s/it]

2018-08-26


 81%|█████████████████████████████████████████████████████████████████▏              | 383/470 [22:02<04:56,  3.40s/it]

2018-09-02


 82%|█████████████████████████████████████████████████████████████████▎              | 384/470 [22:06<04:52,  3.40s/it]

2018-09-09


 82%|█████████████████████████████████████████████████████████████████▌              | 385/470 [22:09<04:48,  3.40s/it]

2018-09-16


 82%|█████████████████████████████████████████████████████████████████▋              | 386/470 [22:12<04:44,  3.39s/it]

2018-09-23


 82%|█████████████████████████████████████████████████████████████████▊              | 387/470 [22:16<04:40,  3.38s/it]

2018-09-30


 83%|██████████████████████████████████████████████████████████████████              | 388/470 [22:19<04:37,  3.39s/it]

2018-10-07


 83%|██████████████████████████████████████████████████████████████████▏             | 389/470 [22:23<04:35,  3.41s/it]

2018-10-14


 83%|██████████████████████████████████████████████████████████████████▍             | 390/470 [22:26<04:33,  3.42s/it]

2018-10-21


 83%|██████████████████████████████████████████████████████████████████▌             | 391/470 [22:29<04:28,  3.40s/it]

2018-10-28


 83%|██████████████████████████████████████████████████████████████████▋             | 392/470 [22:33<04:24,  3.39s/it]

2018-11-04


 84%|██████████████████████████████████████████████████████████████████▉             | 393/470 [22:36<04:21,  3.40s/it]

2018-11-11


 84%|███████████████████████████████████████████████████████████████████             | 394/470 [22:40<04:17,  3.39s/it]

2018-11-18


 84%|███████████████████████████████████████████████████████████████████▏            | 395/470 [22:43<04:15,  3.41s/it]

2018-11-25


 84%|███████████████████████████████████████████████████████████████████▍            | 396/470 [22:46<04:13,  3.42s/it]

2018-12-02


 84%|███████████████████████████████████████████████████████████████████▌            | 397/470 [22:50<04:08,  3.40s/it]

2018-12-09


 85%|███████████████████████████████████████████████████████████████████▋            | 398/470 [22:53<04:08,  3.45s/it]

2018-12-16


 85%|███████████████████████████████████████████████████████████████████▉            | 399/470 [22:57<04:04,  3.44s/it]

2018-12-23


 85%|████████████████████████████████████████████████████████████████████            | 400/470 [23:00<03:58,  3.41s/it]

2018-12-30


 85%|████████████████████████████████████████████████████████████████████▎           | 401/470 [23:03<03:53,  3.39s/it]

2019-01-06


 86%|████████████████████████████████████████████████████████████████████▍           | 402/470 [23:07<03:50,  3.40s/it]

2019-01-13


 86%|████████████████████████████████████████████████████████████████████▌           | 403/470 [23:10<03:49,  3.42s/it]

2019-01-20


 86%|████████████████████████████████████████████████████████████████████▊           | 404/470 [23:14<03:44,  3.40s/it]

2019-01-27


 86%|████████████████████████████████████████████████████████████████████▉           | 405/470 [23:17<03:39,  3.37s/it]

2019-02-03


 86%|█████████████████████████████████████████████████████████████████████           | 406/470 [23:20<03:35,  3.37s/it]

2019-02-10


 87%|█████████████████████████████████████████████████████████████████████▎          | 407/470 [23:24<03:34,  3.40s/it]

2019-02-17


 87%|█████████████████████████████████████████████████████████████████████▍          | 408/470 [23:27<03:31,  3.41s/it]

2019-02-24


 87%|█████████████████████████████████████████████████████████████████████▌          | 409/470 [23:31<03:29,  3.43s/it]

2019-03-03


 87%|█████████████████████████████████████████████████████████████████████▊          | 410/470 [23:34<03:25,  3.43s/it]

2019-03-10


 87%|█████████████████████████████████████████████████████████████████████▉          | 411/470 [23:38<03:21,  3.42s/it]

2019-03-17


 88%|██████████████████████████████████████████████████████████████████████▏         | 412/470 [23:41<03:17,  3.40s/it]

2019-03-24


 88%|██████████████████████████████████████████████████████████████████████▎         | 413/470 [23:44<03:13,  3.39s/it]

2019-03-31


 88%|██████████████████████████████████████████████████████████████████████▍         | 414/470 [23:48<03:08,  3.37s/it]

2019-04-07


 88%|██████████████████████████████████████████████████████████████████████▋         | 415/470 [23:51<03:05,  3.37s/it]

2019-04-14


 89%|██████████████████████████████████████████████████████████████████████▊         | 416/470 [23:54<03:02,  3.38s/it]

2019-04-21


 89%|██████████████████████████████████████████████████████████████████████▉         | 417/470 [23:58<02:58,  3.36s/it]

2019-04-28


 89%|███████████████████████████████████████████████████████████████████████▏        | 418/470 [24:01<02:55,  3.37s/it]

2019-05-05


 89%|███████████████████████████████████████████████████████████████████████▎        | 419/470 [24:05<02:52,  3.38s/it]

2019-05-12


 89%|███████████████████████████████████████████████████████████████████████▍        | 420/470 [24:08<02:49,  3.39s/it]

2019-05-19


 90%|███████████████████████████████████████████████████████████████████████▋        | 421/470 [24:11<02:45,  3.37s/it]

2019-05-26


 90%|███████████████████████████████████████████████████████████████████████▊        | 422/470 [24:15<02:41,  3.37s/it]

2019-06-02


 90%|████████████████████████████████████████████████████████████████████████        | 423/470 [24:18<02:39,  3.40s/it]

2019-06-09


 90%|████████████████████████████████████████████████████████████████████████▏       | 424/470 [24:22<02:37,  3.42s/it]

2019-06-16


 90%|████████████████████████████████████████████████████████████████████████▎       | 425/470 [24:25<02:33,  3.41s/it]

2019-06-23


 91%|████████████████████████████████████████████████████████████████████████▌       | 426/470 [24:28<02:28,  3.37s/it]

2019-06-30


 91%|████████████████████████████████████████████████████████████████████████▋       | 427/470 [24:32<02:24,  3.36s/it]

2019-07-07


 91%|████████████████████████████████████████████████████████████████████████▊       | 428/470 [24:35<02:22,  3.38s/it]

2019-07-14


 91%|█████████████████████████████████████████████████████████████████████████       | 429/470 [24:38<02:18,  3.37s/it]

2019-07-21


 91%|█████████████████████████████████████████████████████████████████████████▏      | 430/470 [24:42<02:15,  3.40s/it]

2019-07-28


 92%|█████████████████████████████████████████████████████████████████████████▎      | 431/470 [24:45<02:12,  3.40s/it]

2019-08-04


 92%|█████████████████████████████████████████████████████████████████████████▌      | 432/470 [24:49<02:08,  3.39s/it]

2019-08-11


 92%|█████████████████████████████████████████████████████████████████████████▋      | 433/470 [24:52<02:05,  3.39s/it]

2019-08-18


 92%|█████████████████████████████████████████████████████████████████████████▊      | 434/470 [24:55<02:02,  3.40s/it]

2019-08-25


 93%|██████████████████████████████████████████████████████████████████████████      | 435/470 [24:59<01:59,  3.42s/it]

2019-09-01


 93%|██████████████████████████████████████████████████████████████████████████▏     | 436/470 [25:02<01:55,  3.41s/it]

2019-09-08


 93%|██████████████████████████████████████████████████████████████████████████▍     | 437/470 [25:06<01:51,  3.39s/it]

2019-09-15


 93%|██████████████████████████████████████████████████████████████████████████▌     | 438/470 [25:09<01:49,  3.43s/it]

2019-09-22


 93%|██████████████████████████████████████████████████████████████████████████▋     | 439/470 [25:12<01:44,  3.39s/it]

2019-09-29


 94%|██████████████████████████████████████████████████████████████████████████▉     | 440/470 [25:16<01:41,  3.38s/it]

2019-10-06


 94%|███████████████████████████████████████████████████████████████████████████     | 441/470 [25:19<01:38,  3.39s/it]

2019-10-13


 94%|███████████████████████████████████████████████████████████████████████████▏    | 442/470 [25:23<01:34,  3.39s/it]

2019-10-20


 94%|███████████████████████████████████████████████████████████████████████████▍    | 443/470 [25:26<01:31,  3.40s/it]

2019-10-27


 94%|███████████████████████████████████████████████████████████████████████████▌    | 444/470 [25:29<01:28,  3.40s/it]

2019-11-03


 95%|███████████████████████████████████████████████████████████████████████████▋    | 445/470 [25:33<01:25,  3.40s/it]

2019-11-10


 95%|███████████████████████████████████████████████████████████████████████████▉    | 446/470 [25:36<01:21,  3.39s/it]

2019-11-17


 95%|████████████████████████████████████████████████████████████████████████████    | 447/470 [25:40<01:18,  3.41s/it]

2019-11-24


 95%|████████████████████████████████████████████████████████████████████████████▎   | 448/470 [25:43<01:14,  3.40s/it]

2019-12-01


 96%|████████████████████████████████████████████████████████████████████████████▍   | 449/470 [25:46<01:11,  3.38s/it]

2019-12-08


 96%|████████████████████████████████████████████████████████████████████████████▌   | 450/470 [25:50<01:07,  3.37s/it]

2019-12-15


 96%|████████████████████████████████████████████████████████████████████████████▊   | 451/470 [25:53<01:03,  3.35s/it]

2019-12-22


 96%|████████████████████████████████████████████████████████████████████████████▉   | 452/470 [25:56<01:00,  3.35s/it]

2019-12-29


 96%|█████████████████████████████████████████████████████████████████████████████   | 453/470 [26:00<00:57,  3.37s/it]

2020-01-05


 97%|█████████████████████████████████████████████████████████████████████████████▎  | 454/470 [26:03<00:53,  3.36s/it]

2020-01-12


 97%|█████████████████████████████████████████████████████████████████████████████▍  | 455/470 [26:06<00:50,  3.37s/it]

2020-01-19


 97%|█████████████████████████████████████████████████████████████████████████████▌  | 456/470 [26:10<00:47,  3.39s/it]

2020-01-26


 97%|█████████████████████████████████████████████████████████████████████████████▊  | 457/470 [26:13<00:43,  3.38s/it]

2020-02-02


 97%|█████████████████████████████████████████████████████████████████████████████▉  | 458/470 [26:17<00:40,  3.35s/it]

2020-02-09


 98%|██████████████████████████████████████████████████████████████████████████████▏ | 459/470 [26:20<00:36,  3.34s/it]

2020-02-16


 98%|██████████████████████████████████████████████████████████████████████████████▎ | 460/470 [26:23<00:33,  3.34s/it]

2020-02-23


 98%|██████████████████████████████████████████████████████████████████████████████▍ | 461/470 [26:27<00:30,  3.35s/it]

2020-03-01


 98%|██████████████████████████████████████████████████████████████████████████████▋ | 462/470 [26:30<00:26,  3.36s/it]

2020-03-08


 99%|██████████████████████████████████████████████████████████████████████████████▊ | 463/470 [26:33<00:23,  3.37s/it]

2020-03-15


 99%|██████████████████████████████████████████████████████████████████████████████▉ | 464/470 [26:37<00:20,  3.40s/it]

2020-03-22


 99%|███████████████████████████████████████████████████████████████████████████████▏| 465/470 [26:40<00:16,  3.38s/it]

2020-03-29


 99%|███████████████████████████████████████████████████████████████████████████████▎| 466/470 [26:44<00:13,  3.42s/it]

2020-04-05


 99%|███████████████████████████████████████████████████████████████████████████████▍| 467/470 [26:47<00:10,  3.42s/it]

2020-04-12


100%|███████████████████████████████████████████████████████████████████████████████▋| 468/470 [26:50<00:06,  3.41s/it]

2020-04-19


100%|███████████████████████████████████████████████████████████████████████████████▊| 469/470 [26:54<00:03,  3.38s/it]

2020-04-26


100%|████████████████████████████████████████████████████████████████████████████████| 470/470 [26:57<00:00,  3.44s/it]


Because the Times API has a limit of maximum 20 calls per minute with 3 seconds lapses between calls it takes time for the module to make all the given calls.

* In our case it took 27 minutes to make 470 calls.

-----------
Now I will extract only the dictionaries containing the data about bestseller books.
In order to do that I use personally defined function `converter` which will extract values based on given keys.

In [3]:
keys = ['weeks_on_list', 'primary_isbn10', 'primary_isbn13',
       'publisher', 'description', 'title', 'author']

In [4]:
def converter(dic, keys):
    converted = {}
    for n in range(0, len(dic)):
        for m in range(0, len(dic[n])):
            for k in keys:
                converted.setdefault(k, []).append(dic[n][m][k])
    return converted

In [5]:
# Calling function with extracted data from the Times API and hand picked keys
bbb = converter(dic, keys)

Finally, I convert the dictionary into the `pandas` dataframe before saving it.

In [6]:
df = pd.DataFrame.from_dict(bbb)
df.head()

weeks_on_list primary_isbn10 primary_isbn13      publisher  \
0             12           None  9781616200817      Algonquin   
1              1           None  9781101513781  Penguin Group   
2              6     0759514712  9780759514713  Little, Brown   
3              2     0316069388  9780316069380  Little, Brown   
4              9     0425232204  9780425232200  Penguin Group   

                                         description                title  \
0  After his parents die in a car accident, a you...  WATER FOR ELEPHANTS   
1  A smoke jumper faces a new season of firefight...         CHASING FIRE   
2  Routinely doing business from his Lincoln Town...   THE LINCOLN LAWYER   
3  The defense lawyer Mickey Haller represents a ...    THE FIFTH WITNESS   
4  A young white woman and two black maids in 196...             THE HELP   

             author  
0        Sara Gruen  
1      Nora Roberts  
2  Michael Connelly  
3  Michael Connelly  
4  Kathryn Stockett

In [11]:
df['best_seller'] = 'yes'

In [12]:
print('dimensions: ', df.shape)

dimensions:  (8555, 8)


We have extracted 8555 rows and seven columns from the Times API containing books that have been on their best-sellers list.

Final step: I save the dataframe to the pickle file for more convenient transferring into the notebook where I will perform additional work on the dataframe.

In [13]:
# df.to_csv('best_sellers.csv', index=False)
df.to_pickle('best_sellers.pkl')